parsing/syntaxerr.ml

```diff
@@ -28,32 +28,35 @@ let prepare_error = function
   | Unclosed(opening_loc, opening, closing_loc, closing) ->
       Location.errorf ~loc:closing_loc
         ~sub:[
-          Location.error ~loc:opening_loc
-            (Printf.sprintf "Error: This '%s' might be unmatched" opening)
+          Location.errorf ~loc:opening_loc
+            "%a This '%s' might be unmatched" Location.print_error_prefix () opening
```

Comment:
Could we export a `Location.errorf_prefixed` to factor these `Location.print_error_prefix`? There are a couple of calls to `Location.errorf` in the compiler without the `Error:` prefix (I see one in `ctype.ml` and three in `typetexp.ml`), and you should feel free to convert all of them to `errorf_prefixed` (it is inconsistent that they don't use `Error:`). Users of `raise_errorf` is a slightly different matter.